snuba/clickhouse/native.py

```diff
@@ -240,6 +240,23 @@ def execute(
                             # balancer a chance to mark a bad host as down.
                             time.sleep(0.1)
                 except errors.Error as e:
+                    if e.code == errors.ErrorCodes.TOO_MANY_SIMULTANEOUS_QUERIES:
```

Comment:
How about we fetch the `simultaneous_queries_sleep_seconds` from below and as part of the conditional here make sure that it's not None? That gives us a simple kill-switch if we're retrying and causing more overload

If it's None we allow 0 retries -- if it's any value we continue into this clause.

Reply:
Sure.